projects/lib/schematics/make-static/rules/config-rule.ts

```diff
@@ -0,0 +1,195 @@
+import { Tree, Rule, SchematicsException } from "@angular-devkit/schematics";
+import ts, {factory} from "typescript";
+import { MakeStaticOptions } from "../schema";
+
+/**
+ * `LiteralValueType` is a type alias that represents the possible types of
+ * literal values in TypeScript, such as strings, numbers, and booleans. It is
+ * used in the `createProperty` and `createValue` functions to ensure that the
+ * correct type of value is created for a given property assignment.
+ */
+type LiteralValueType = string | number | boolean;
+
+let CONFIG_IDENTIFIER = 'GLOBAL_CONFIG';
+
+/**
+ * This variable is later used to store
+ * the configuration data read from a JSON file and to update an existing
+ * TypeScript object literal expression or create a new one.
+ */
+let configuration = {};
+
+/**
+ * This function reads a configuration file and updates a TypeScript source file
+ * with a custom transformer or creates a new config object literal if it doesn't
+ * exist.
+ * @param options - An object containing options for the function, including:
+ * @returns A function that takes in an options object and returns a Rule.
+ */
+export function addConfigObjectRule(options: MakeStaticOptions): Rule {
+  return async (tree: Tree) => {
+
+    // exit when 'config-flag' is not provided
+    if (!options?.configPath) return;
+
+    CONFIG_IDENTIFIER = (options.configIdentifier || CONFIG_IDENTIFIER)
+      .replaceAll('-', '_')
+      .toUpperCase();
+
+    // first read the configuration json file
+    const configFileContent = tree.read(options.config);
+    configuration = JSON.parse(configFileContent?.toString() || '');
+    if (!Array.isArray(configuration)) {
+      throw new SchematicsException(`Expected an array of configuration objects in ${options.config}: ${configuration}`);
+    }
+    // keep only 'global' object and remove 'id' and 'type' useless
+    [configuration] = configuration.filter(c => c.type === "global");

```

Comment:
This is very specific to vanilla builder, but in the library it might be useful to write all the configuration (including id and type).